src/misc/lv_event.c

```diff
@@ -153,19 +172,19 @@ bool lv_event_remove(lv_event_list_t * list, uint32_t index)
 {
     LV_ASSERT_NULL(list);
     lv_event_dsc_t * dsc = lv_event_get_dsc(list, index);
-    lv_free(dsc);
-    return lv_array_remove(list, index);
+    if(dsc == NULL) return false;
+    event_mark_deleting(list, dsc);
+    cleanup_event_list(list);
+    return true;
 }
 
 void lv_event_remove_all(lv_event_list_t * list)
 {
     LV_ASSERT_NULL(list);
-    int size = lv_array_size(list);
-    lv_event_dsc_t ** dsc = lv_array_front(list);
-    for(int i = 0; i < size; i++) {
-        lv_free(dsc[i]);
-    }
-    lv_array_deinit(list);
+    int size = event_array_size(list);
+    for(int i = 0; i < size; i++)
+        event_mark_deleting(list, *event_array_at(list, i));
+    cleanup_event_list(list);
```

Comment:
Since now the array won't be changed during iteration, we can use the pointers directly?
```c
    int size = lv_array_size(list);
    lv_event_dsc_t ** events = event_array_at(list, 0);
    for(int i = 0; i < size; i++) {
        event_mark_deleting(list, &events[i]); // no need to call event_array_at to check if i is valid?
    }
    cleanup_event_list(list); 
```